wsw/fingerprint.py

```diff
@@ -4,120 +4,114 @@
 import matplotlib.pyplot as plt
 
 
-def lpfilter(signal, sr):
-    cutoff = 10e3
-    sos = sig.butter(10, cutoff, fs=sr, btype='lowpass', analog=False, output='sos')
-    return sig.sosfilt(sos, signal)
-
-
-def downsample(signal, old_sr, new_sr):
-    return librosa.resample(signal, old_sr, new_sr)
-
-
-def stft(signal, N):
-    spec = librosa.stft(signal, n_fft=N, window=sig.windows.hamming)
-    return np.abs(spec)
-
-
-def denoise(spectrogram):
-    return librosa.decompose.nn_filter(spectrogram)
-
-
-def log_bin(arr, n_bins):
-    bands = np.array([10 * 2 ** i for i in range(n_bins - 1)])
-    idxs = np.arange(len(arr))
-    split_arr = np.split(arr, np.searchsorted(idxs, bands))
-    return split_arr
-
-
-def ft_filter(ft, n_bins, alpha=1):
-    binned_ft = log_bin(ft, n_bins)
-    for i, band in enumerate(binned_ft):
-        band_max = np.max(band)
-        filtered = np.where(band < band_max, 0, band)
-        binned_ft[i] = filtered
-    joined_filtered = np.concatenate(binned_ft)
-    thresh = np.mean(joined_filtered)
-
-    return np.where(joined_filtered < thresh * alpha, 0, joined_filtered)
-
-
-def spec_filter(spec, n_bins):
-    filtered = np.zeros_like(spec.T)
-    for i, dft in enumerate(spec.T):
-        filtered[i] = ft_filter(dft, n_bins)
-    return filtered.T
-
-
 class Fingerprint:
-    def __init__(self, path, sr=22050, n_fft=512):
-        y, sr = librosa.load(path, sr)
+    """
+    This is a simple class that contains class variables describing audio data. These
+    class variables can be thought of as 'fingerprints' of the original audio signal.
+    Currently, the fingerprint class stores two forms of the data: Original signal and
+    a filtered spectrogram which is a sparse matrix describing the most dominant frequencies
+    of the original signal.
+
+    :param audio_path: The path from which to load audio data for fingerprinting
+    :param sr: The desired sample rate to load audio data at
+    :param n_fft: The number of DFTs to use in creating the STFT/spectrogram
+        fingerprint of the original audio data.
+
+
+    """
+    def __init__(self, audio_path, sr=22050, n_fft=512):
+        y, sr = librosa.load(audio_path, sr)
         self.signal = y
         self.sr = sr
-        self.n_ftt = 512
+        self.n_ftt = n_fft
```

Comment:
`self.n_ftt` should be `self.n_fft`.  Just a nitpick.

Reply:
Got it!